TesseractOCR/G8Tesseract.mm

```diff
@@ -159,43 +194,57 @@ - (BOOL)resetEngine
     return isInitDone;
 }
 
-- (void)copyDataToDocumentsDirectory
+- (BOOL)moveTessdataToCachesDirectoryIfNecessary
 {
-    // Useful paths
     NSFileManager *fileManager = [NSFileManager defaultManager];
-    NSArray *documentPaths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
-    NSString *documentPath = documentPaths.firstObject;
-    NSString *dataPath = [documentPath stringByAppendingPathComponent:self.dataPath];
-
-    //	NSString *dataPath = [[NSBundle mainBundle] pathForResource:@"grc" ofType:@"traineddata"];
-    NSLog(@"DATAPATH %@", dataPath);
-
-    // Copy data in Doc Directory
-    if ([fileManager fileExistsAtPath:dataPath] == NO) {
-        [fileManager createDirectoryAtPath:dataPath withIntermediateDirectories:YES attributes:nil error:nil];
+    
+    // Useful paths
+    NSString *tessdataFolderName = @"tessdata";
+    NSString *tessdataPath = [[NSBundle bundleForClass:self.class].resourcePath stringByAppendingPathComponent:tessdataFolderName];
+    NSString *destinationPath = [self.absoluteDataPath stringByAppendingPathComponent:tessdataFolderName];
+    NSLog(@"Tesseract destination path: %@", destinationPath);
+    
+    if ([fileManager fileExistsAtPath:destinationPath] == NO) {
+        NSError *error = nil;
+        [fileManager createDirectoryAtPath:destinationPath withIntermediateDirectories:YES attributes:nil error:&error];
+        if (error != nil) {
+            NSLog(@"Error creating folder %@: %@", destinationPath, error);
+            return NO;
+        }
     }
-
-    NSBundle *bundle = [NSBundle bundleForClass:[self class]];
-    for (NSString *languageName in [self.language componentsSeparatedByString:@"+"]) {
-        NSString *tessdataPath = [bundle pathForResource:languageName ofType:@"traineddata"];
-
-        if (tessdataPath != nil) {
-            NSString *destinationPath = [dataPath stringByAppendingPathComponent:tessdataPath.lastPathComponent];
-
-            if([fileManager fileExistsAtPath:destinationPath] == NO) {
-                NSError *error = nil;
-                NSLog(@"found %@", tessdataPath);
-                NSLog(@"coping in %@", destinationPath);
-                [fileManager copyItemAtPath:tessdataPath toPath:destinationPath error:&error];
-
-                if(error != nil) {
-                    NSLog(@"ERROR! %@", error.description);
-                }
+    
+    BOOL result = YES;
+    NSError *error = nil;
+    NSArray *files = [fileManager contentsOfDirectoryAtPath:tessdataPath error:&error];
```

Comment:
Yes I agree that if the directory is **empty** then the function will return nil, but what I'm talking about is if the directory **does not exist**. What will the function do then?